tests/Microsoft.Azure.Batch.SoftwareEntitlement.Tests/NodeEntitlementsBuilderTests.cs

```diff
@@ -292,5 +324,85 @@ public void WhenValid_PropertyHasExpectedValue()
                 result.Value.Issuer.Should().Be(_issuer);
             }
         }
+
+        public class BatchAccountIdProperty : NodeEntitlementsBuilderTests
+        {
+            [Fact]
+            public void WhenMissing_BuildReturnsNull()
+            {
+                _commandLine.BatchAccountId = null;
+                var result = NodeEntitlementsBuilder.Build(_commandLine);
+                result.Value.BatchAccountId.Should().BeNull();
+            }
+
+            [Fact]
+            public void WhenSpecified_PropertyHasExpectedValue()
+            {
+                const string specifiedBatchAccountId = "testbatchaccountid";
```

Comment:
I'd move this constant out onto the class because the exact value used isn't important for the test.

Reply:
Happy to move it for consistency. For the record I have a slight preference for keeping it in the test, just because that makes feel more comfortable when reading tests that I completely understand what's going on. Whenever a test refers to a field outside of the test itself I feel the need to check what interactions other code might be having with that field (obviously not much if it's a constant, but I still have to navigate to the declaration to see that).

Reply:
Understood, it's a trade-off. I like to keep the tests themselves as simple as possible and having the same constant defined in multiple tests just feels like noise.